admin/admin-nav-menu.php

```diff
@@ -262,14 +273,22 @@ public function theme_mod_nav_menu_locations( $menus ) {
 	 * @return void
 	 */
 	public function delete_nav_menu( $term_id ) {
-		foreach ( $this->options['nav_menus'] as $theme => $locations ) {
+		if ( empty( $this->options->get( 'nav_menus' ) ) ) {
```

Comment:
This is to prevent calling `set()` with empty value?

Reply:
Yes, to prevent calling `set()` while there is nothing to update.